vc/vc.go

```diff
@@ -0,0 +1,198 @@
+package vc
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/tbd54566975/web5-go/dids/did"
+	"github.com/tbd54566975/web5-go/jwt"
+)
+
+// these constants are defined in the W3C Verifiable Credential Data Model specification for:
+//   - [Context]
+//   - [Type]
+//
+// [Context]: https://www.w3.org/TR/vc-data-model/#contexts
+// [Type]: https://www.w3.org/TR/vc-data-model/#dfn-type
+const (
+	BaseContext = "https://www.w3.org/2018/credentials/v1"
+	BaseType    = "VerifiableCredential"
+)
+
+// DataModel represents the W3C Verifiable Credential Data Model defined [here]
+//
+// [here]: https://www.w3.org/TR/vc-data-model/
+type DataModel[T CredentialSubject] struct {
+	Context           []string `json:"@context"`                 // https://www.w3.org/TR/vc-data-model/#contexts
+	Type              []string `json:"type"`                     // https://www.w3.org/TR/vc-data-model/#dfn-type
+	Issuer            string   `json:"issuer"`                   // https://www.w3.org/TR/vc-data-model/#issuer
+	CredentialSubject T        `json:"credentialSubject"`        // https://www.w3.org/TR/vc-data-model/#credential-subject
+	ID                string   `json:"id,omitempty"`             // https://www.w3.org/TR/vc-data-model/#identifiers
+	IssuanceDate      string   `json:"issuanceDate"`             // https://www.w3.org/TR/vc-data-model/#issuance-date
+	ExpirationDate    string   `json:"expirationDate,omitempty"` // https://www.w3.org/TR/vc-data-model/#expiration
```

Comment:
we should include credential schema and evidence here, I believe. even if it's only 'right' to use it with v2, both are allowed by v1 and we may need both

Reply:
incoming

Reply:
opening issues to include both of these.